grpc-api/src/main/java/io/deephaven/grpc_api/session/SessionService.java

```diff
@@ -51,6 +57,64 @@ public SessionService(final Scheduler scheduler, final SessionState.Factory sess
 
         // Protect ourselves from rotation spam, but be loose enough that any reasonable refresh strategy works.
         this.tokenRotateMs = tokenExpireMs / 5;
+
+        if (ProcessEnvironment.tryGet() != null) {
+            ProcessEnvironment.getGlobalFatalErrorReporter().addInterceptor(this::onFatalError);
+        }
+    }
+
+    private synchronized void onFatalError(
+            @NotNull String message,
+            @NotNull Throwable throwable,
+            boolean isFromUncaught) {
+        final TerminationNotificationResponse.Builder builder =
+                TerminationNotificationResponse.newBuilder()
+                        .setAbnormalTermination(true)
+                        .setIsFromUncaughtException(isFromUncaught)
+                        .setReason(message);
+
+        // TODO (core#801): revisit this error communication to properly match the API Error mode
+        while (throwable != null) {
+            builder.addStackTraces(transformToProtoBuf(throwable));
+            throwable = throwable.getCause();
+        }
+
+        final TerminationNotificationResponse notification = builder.build();
+        terminationListeners.forEach(listener -> listener.sendMessage(notification));
+        terminationListeners.clear();
+    }
+
+    private static TerminationNotificationResponse.StackTrace transformToProtoBuf(@NotNull final Throwable throwable) {
+        return TerminationNotificationResponse.StackTrace.newBuilder()
+                .setType(throwable.getClass().getSimpleName())
```

Comment:
my mild preference would be that if we only send the simple name, make sure we qualify it with any enclosing element other than a package, to avoid Foo.Error vs Bar.Error.

But if we're already shipping the full stack trace with full frames (and each frame has the fully qualified class name), we should just suck it up and put the fqcn here too